runtime/vm/createramclass.cpp

```diff
@@ -1439,15 +1439,37 @@ checkPackageAccess(J9VMThread *vmThread, J9Class *foundClass, UDATA classPreload
 }
 
 /**
- * Sets the current exception using the specified class name.
+ * Sets the current exception using the detailed error message plus the specified class name.
  */
 static void
-setCurrentExceptionForBadClass(J9VMThread *vmThread, J9UTF8 *badClassName, UDATA exceptionIndex)
+setCurrentExceptionForBadClass(J9VMThread *vmThread, J9UTF8 *badClassName, UDATA exceptionIndex, U_32 nlsModuleName, U_32 nlsMessageID)
 {
-	J9MemoryManagerFunctions *gcFuncs = vmThread->javaVM->memoryManagerFunctions;
-	j9object_t detailMessage = gcFuncs->j9gc_createJavaLangString(vmThread, J9UTF8_DATA(badClassName), J9UTF8_LENGTH(badClassName), J9_STR_XLAT);
+	PORT_ACCESS_FROM_VMC(vmThread);
+	char * errorMsg = NULL;
+	const char * nlsMessage = OMRPORT_FROM_J9PORT(PORTLIB)->nls_lookup_message(
+			OMRPORT_FROM_J9PORT(PORTLIB),
+			J9NLS_DO_NOT_PRINT_MESSAGE_TAG | J9NLS_DO_NOT_APPEND_NEWLINE,
+			nlsModuleName, nlsMessageID,
+			NULL);
+
+	if (NULL != nlsMessage) {
+		U_16 badClassNameLength = J9UTF8_LENGTH(badClassName);
+		U_8 * badClassNameStr = J9UTF8_DATA(badClassName);
+
+		UDATA errorMsgLen = j9str_printf(PORTLIB, NULL, 0, nlsMessage, badClassNameLength, badClassNameStr);
+		errorMsg = (char*)j9mem_allocate_memory(errorMsgLen, OMRMEM_CATEGORY_VM);
+		if (NULL == errorMsg) {
+			J9MemoryManagerFunctions *gcFuncs = vmThread->javaVM->memoryManagerFunctions;
+			j9object_t detailMessage = gcFuncs->j9gc_createJavaLangString(vmThread, badClassNameStr, badClassNameLength, J9_STR_XLAT);
+			setCurrentException(vmThread, exceptionIndex, (UDATA *)detailMessage);
+			setNativeOutOfMemoryError(vmThread, 0, 0);
```

Comment:
We can't set both exceptions here as the second call will overwrite the first.  Delete the `setNativeOutOfMemoryError` call and only do `setCurrentException`

Reply:
Already removed `setNativeOutOfMemoryError`.